WebContent/active/Unvalidated-Redirect/Redirect-FalsePositives-GET/inclusion-logic3.jsp

```diff
@@ -282,9 +283,24 @@
                     	}
                     					
                     	if (defaultInputType == DefaultInputType.RELATIVE_INPUT) {
+                            // This had previously been using request.getServerName().
+                            // However, that actually means that it 'trusts' the Host
+                            // header from the traffic, which turns this test which was
+                            // meant to be a FP case into a TP case for scenarios when
+                            // headers are being manipulated by a scanner.
+                            // There should only be one configured (default is localhost)
+                            MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
+                            ObjectName engineObj = new ObjectName("Catalina:type=Engine,*");
+                            Set<ObjectName> engines = mbs.queryNames(engineObj, null);
+                            String defaultHost = "";
+                            if (!engines.isEmpty()) {
+                                ObjectName engine = engines.iterator().next();
+                                defaultHost = (String) mbs.getAttribute(engine, "defaultHost");
+                            }
+
                     		//relative to current dir path
                     		defaultBasePath = FileConstants.HTTP_PREFIX
-                           		+ request.getServerName() + ":" +
+                           		+ defaultHost + ":" +
                     			request.getServerPort() + request.getContextPath() + 
```

Comment:
fwiw the port is also obtained from the host header

Reply:
I don't mind either way, while it leads to different port it does not matter for the FP.